src/itemlistformaction.cpp

```diff
@@ -486,13 +486,20 @@ bool ItemListFormAction::process_operation(Operation op,
 		try {
 			const auto message_lifetime = v->get_statusline().show_message_until_finished(
 					_("Marking feed read..."));
+
 			std::vector<std::string> guids;
 			for (const auto& item : visible_items) {
 				const std::string guid = item.first->guid();
 				guids.push_back(guid);
 			}
 			rsscache->mark_items_read_by_guid(guids);
-			v->get_ctrl()->mark_all_read(guids);
+
+			if (apply_filter) {
```

Comment:
We need to make sure to update this if we ever add some other way to filter items (otherwise, items could be marked as read on the remote end, without being updated locally):
https://github.com/newsboat/newsboat/blob/c0ee18a17526c6c5f35b11fd48a49ff55bd0f72e/src/itemlistformaction.cpp#L932-L946

However, looks correct/good now